dist/index.js

```diff
@@ -7350,63 +7350,93 @@ async function getOTPVersion(otpSpec0, osVersion) {
 }
 
 async function getElixirVersion(exSpec0, otpVersion0) {
-  const otpVersion = otpVersion0.match(/^([^-]+-)?(.+)$/)[2]
-  const otpVersionMajor = otpVersion.match(/^([^.]+).*$/)[1]
-  const elixirVersions = await getElixirVersions()
-  const semverVersions = Array.from(elixirVersions.keys()).sort()
-  const exSpec = exSpec0.replace(/-otp-.*$/, '')
-  const elixirVersionFromSpec = getVersionFromSpec(exSpec, semverVersions, true)
-  let elixirVersionForDownload = elixirVersionFromSpec
-  if (isVersion(otpVersionMajor)) {
-    elixirVersionForDownload = `${elixirVersionFromSpec}-otp-${otpVersionMajor}`
-  }
-  if (elixirVersionFromSpec === null) {
-    throw new Error(
-      `Requested Elixir version (${exSpec0}) not found in version list ` +
-        "(should you be using option 'version-type': 'strict'?)",
-    )
-  }
-
-  const elixirVersionComp = elixirVersions.get(elixirVersionFromSpec)
-  if (
-    (elixirVersionComp && elixirVersionComp.includes(otpVersionMajor)) ||
-    !isVersion(otpVersionMajor)
-  ) {
-    core.info(
-      `Using Elixir ${elixirVersionFromSpec} (built for Erlang/OTP ${otpVersionMajor})`,
+  let elixirVersionForDownload
+  if (isSemVer(exSpec0)) {
```

Comment:
Hmm, this won't actually work well, that is, if we put this here, then we opt of compatibility checks. Maybe we skip doing this for elixir, not sure.

Reply:
Could we maybe go with isStrictVersion, in this case? Or all cases? I'm afraid we're already "guessing" to much. e.g. 0.23 (Gleam) is not SemVer but it is a valid final version, as is Elixir's 1.14, and OTP's 25.1.

Reply:
(I guess this is what's making tests fail for this pull request - ?)

Reply:
I think you are right, and yes, we would have to go with isStrictVersion, though doesn't help @lpil for his case.